service-base/src/main/java/org/eclipse/hono/service/HealthCheckRegistration.java

```diff
@@ -0,0 +1,28 @@
+/*******************************************************************************
+ * Copyright (c) 2016, 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.service;
+
+/**
+ * Provides a server for health checks.
+ */
+public interface HealthCheckRegistration extends Lifecycle {
```

Comment:
I wonder why this interface isn't called `HealthCheckServer`. Isn't that what it is supposed to be (according to the JavaDoc)?

Reply:
To be honest, this was a compromise because that is already the name of the class implementing it and the diff would be horror if I rename the class and give it's name to the interface. If you are willing to go through diff hell I can rename to rename both :)

Reply:
I do not see the issue, there are only very few places where HealthCheckServer is used. SO FMPOV we should rename the existing `HealthCheckServer` class to e.g. `VertxBasedHealthCheckServer` and rename `HealthCheckRegistration` to `HealthCheckServer`.

Reply:
There is no issue with the code, just with the tooling. The diff shown in github and probably every IDE will compare the new interface with the old implementation class, because they have the same name, making it harder for you to spot the difference in the code. But no problem, I will change it.